components/two-factor-authentication/AuthenticatorSettings.tsx

```diff
@@ -0,0 +1,275 @@
+              <UserTwoFactorMethodItem individual={props.individual} userTwoFactorMethod={device} />
+            </Box>
+          );
+        })}
+      </Box>
+      <Box mt={3}>
+        <StyledButton onClick={() => setIsAddingAuthenticator(true)} buttonSize="tiny" buttonStyle="secondary">
+          <FormattedMessage defaultMessage="Add authenticator" /> <PlusIcon size="14px" />
+        </StyledButton>
+      </Box>
+      {isAddingAuthenticator && (
+        <AddAuthenticatorModal
+          onClose={() => setIsAddingAuthenticator(false)}
+          individual={props.individual}
+          onRecoveryCodes={props.onRecoveryCodes}
+        />
+      )}
+    </StyledCard>
+  );
+}
+
+type AddAuthenticatorModalProps = {
+  individual: Pick<Individual, 'id' | 'email'>;
+  onRecoveryCodes: (codes: string[]) => void;
+  onClose: () => void;
+};
+
+function AddAuthenticatorModal(props: AddAuthenticatorModalProps) {
+  const { addToast } = useToasts();
+  const intl = useIntl();
+
+  const { otpAuthUrl, base32 } = React.useMemo(() => {
+    return generateNewAuthenticatorAppSecret(props.individual.email);
+  }, [props.individual.email]);
+
+  const [addAuthenticatorAppMutation] = useMutation<{
+    addTwoFactorAuthTokenToIndividual: {
+      account: Pick<Individual, 'twoFactorMethods'>;
+      recoveryCodes: string[];
+    };
+  }>(AddAuthenticatorAppToIndividualMutation, {
+    context: API_V2_CONTEXT,
+    variables: {
+      account: {
+        id: props.individual.id,
+      },
+      token: base32,
+    },
+  });
+
+  const formik = useFormik({
+    initialValues: {
+      twoFactorAuthenticatorCode: '',
+    },
+    async onSubmit() {
+      try {
+        const result = await addAuthenticatorAppMutation();
+        addToast({
+          type: TOAST_TYPE.SUCCESS,
+          message: <FormattedMessage defaultMessage="Authenticator added" />,
+        });
+
+        if (result.data.addTwoFactorAuthTokenToIndividual.recoveryCodes) {
+          props.onRecoveryCodes(result.data.addTwoFactorAuthTokenToIndividual.recoveryCodes);
+        }
+      } catch (e) {
+        addToast({
+          type: TOAST_TYPE.ERROR,
+          message: i18nGraphqlException(intl, e),
+        });
+      } finally {
+        props.onClose();
+      }
+    },
+    validate(values) {
+      const errors: Record<string, unknown> = {};
+      if (!values.twoFactorAuthenticatorCode) {
+        errors.twoFactorAuthenticatorCode = intl.formatMessage(I18nMessages.REQUIRED);
+      } else {
+        const verified = speakeasy.totp.verify({
+          secret: base32,
+          encoding: 'base32',
+          token: values.twoFactorAuthenticatorCode,
+          window: 2,
+        });
+
+        if (!verified) {
+          errors.twoFactorAuthenticatorCode = intl.formatMessage(I18nMessages.INVALID_TOTP_CODE);
+        }
+      }
+
+      return errors;
+    },
+  });
+
+  return (
+    <StyledModal onClose={props.onClose}>
+      <Box>
+        <Flex gap="20px">
```

Comment:
This modal is missing a header/title:
![image](https://github.com/opencollective/opencollective-frontend/assets/2119706/af08cc28-fa53-46c3-9488-e196a7cfe640)

It would also be cool to move the code input beneath the token box.